app/src/org/commcare/fragments/personalId/PersonalIdPhoneVerificationFragment.java

```diff
@@ -342,11 +351,38 @@ private void navigateToNameEntry() {
     }
 
     @Override
-    protected void navigateToMessageDisplay(@NotNull String title,
-            @org.jetbrains.annotations.Nullable String message, boolean isCancellable, int phase, int buttonText) {
+    protected void navigateToMessageDisplay(
+            @NotNull String title,
+            @org.jetbrains.annotations.Nullable String message,
+            boolean isCancellable,
+            int phase,
+            int buttonText
+    ) {
         NavDirections directions = PersonalIdPhoneVerificationFragmentDirections
                 .actionPersonalidOtpPageToPersonalidMessage(title, message, phase, getString(buttonText),
                         null).setIsCancellable(isCancellable);
         Navigation.findNavController(binding.getRoot()).navigate(directions);
     }
+
+    private void setupOtpManagerForReattempt() {
+        int otpReattempts = personalIdSessionData.getOtpReattempts();
+
+        // Fallback to Twilio (via Personal ID) if this is the first time the user reattempts to send the OTP.
+        if (otpReattempts == 0) {
+            otpManager = new OtpManager(
+                    activity,
+                    personalIdSessionData,
+                    otpCallback,
+                    SMS_METHOD_PERSONAL_ID
+            );
```

Comment:
- this also need to check on `personalIdSessionData.getOtpFallback()`  similar to `shouldAutoSwitchToPersonalIdAuth` . That will ensure that we are only switching to this when Personal ID fallback is allowed for this user

Reply:
That's good to know, will do!